docs/tests.md

```diff
@@ -0,0 +1,439 @@
+    - this two tests might be generalized in a way and merged into one, the only difference is a resource type
+
+This test checks that webhook configurations are presented in cluster then delete it. Waits for configurations to be restored and 
+being equal to what is encoded within the vendored MAPO repo.
+
+This test checks operator behaviour to manage and maintain MAPI webhooks state.
+
+#### Machine API operator deployment should maintains spec after mutating webhook configuration change and preserve caBundle
+#### Machine API operator deployment should maintains spec after validating webhook configuration change and preserve caBundle
+* Parallel: No (each)
+* ExecTime: ~5s on AWS (each)
+* Recommendation:
+    - break dependency to the vendored MAPO if it's possible
+    - this two tests might be generalized in a way and merged into one, the only difference is a resource type
+
+This test checks that webhook configurations are presented in cluster then changes it. Waits for configurations to be restored and
+being equal to what is encoded within the vendored MAPO repo.
+
+This test checks operator behaviour to manage and maintain MAPI webhooks state.
+
+#### Machine API cluster operator status should be available
+* Parallel: Yes
+* ExecTime: ~2s on AWS
+
+Checks cluster operator status to be available, not progressing and not degraded.
+
+#### When cluster-wide proxy is configured, Machine API cluster operator should  create machines when configured behind a proxy
+* Parallel: No
+* ExecTime: ~30m on AWS
+* Recommendation:
+  - ~make this test periodic (done already)~
+  - extend this test - add check for the CCCMO also
+
+This test consists of several stages and exercises MAPO ability to pick up and properly handle cluster-wide proxy settings as well as
+machine-controllers ability to operate behind a proxy server:
+
+1) Deploying HTTP proxy server as a DeamonSet
+2) Waits for machine-api controllers will be redeployed with updated proxy settings
+3) Creates machineset, waits when machine will be created and hit the Running phase
+4) Removes proxy
+5) Waits for machine-api controllers will be redeployed without proxy settings
+
+Since configuring proxy causes deployment updates for KCM, MAO and a bunch of other system critical components, this test takes quite long time.
+Also, this test is quite disruptive and should be running in Serial mode, due to it touches cluster-wide settings which affect not only MAO parts.
+
+### Cluster Machine Approver
+
+Recommendation: This suite needs to be extended with some sort of disruptive suite (change/delete CMA deployment, ensure that it's preserved).
+Another variant is to remove this test at all since a cluster will not finish installation if the operator will be in the degraded state.
+
+#### Cluster Machine Approver deployment should be available
+* Parallel: Yes
+* ExecTime: ~2s on AWS
+
+This test checks status of the `openshift-cluster-machine-approver/machine-approver` deployment.
```

Comment:
Yup, might be. 

This might be merged with another CMA test, which checks the deployment. Aside from that two, we do not have any specific CMA tests

Reply:
Or yes, pre-step. Will be convenient if we will decide to add another CMA-related tests.